inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/openapi/DataProxyController.java

```diff
@@ -80,4 +82,73 @@ public String getAllConfig(@RequestParam String clusterName, @RequestParam(requi
         return clusterService.getAllConfig(clusterName, md5);
     }
 
+    /**
+     * changeClusterTag
+     */
+    @RequestMapping(value = "/changeClusterTag", method = RequestMethod.POST)
+    @ApiOperation(value = "Change cluster tag and topic of a inlong group id.")
+    public Response<String> changeClusterTag(@RequestBody InlongGroupEntity groupInfo) {
+        String inlongGroupId = groupInfo.getInlongGroupId();
+        String clusterTag = groupInfo.getInlongClusterTag();
+        String topic = groupInfo.getMqResource();
+        if (StringUtils.isEmpty(inlongGroupId) || StringUtils.isEmpty(clusterTag) || StringUtils.isEmpty(topic)) {
+            throw new BusinessException(ErrorCodeEnum.GROUP_NOT_FOUND);
+        }
+        // select
+        InlongGroupEntity oldGroup = inlongGroupMapper.selectByGroupId(inlongGroupId);
+        if (oldGroup == null) {
+            throw new BusinessException(ErrorCodeEnum.GROUP_NOT_FOUND);
+        }
+        // parse ext_params
+        String extParams = oldGroup.getExtParams();
+        if (StringUtils.isEmpty(extParams)) {
+            extParams = "{}";
+        }
+        // parse json
+        Gson gson = new Gson();
```

Comment:
Suggest using a global static instance of `Gson`.

Reply:
It is not a promise that Gson is a thread safe class.
DataProxyController have synchronized operation.